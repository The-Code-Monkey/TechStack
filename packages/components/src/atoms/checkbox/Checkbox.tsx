import { Interactable, Icon } from '../../primatives';
import { generateAutomationId } from '../../utils';

import { disabledStyles, checkedStyles } from './styled';

export interface Props {
  autoid?: string;
  onClick?: (p1: boolean, p2: MouseEvent) => void;
  checked?: boolean;
  indeterminate?: boolean;
  disabled?: boolean;
}

const Checkbox = ({
  indeterminate,
  checked,
  onClick,
  disabled,
  autoid,
}: Props) => {
  const handleOnClick = (event: MouseEvent) => {
    const status = indeterminate || !checked;

    if (onClick) {
      onClick(status, event);
    }
  };

  return (
    <Interactable
      display='flex'
      size={7}
      alignItems='center'
      justifyContent='center'
      cursor='pointer'
      bgColor='neutrals.0'
      border='1'
      borderRadius='1'
      borderColor='neutrals.10'
      autoid={autoid ? generateAutomationId(`${autoid}_checkbox`) : 'checkbox'}
      data-checked={checked}
      data-indeterminate={indeterminate}
      data-disabled={disabled}
      onClick={handleOnClick}
      {...(disabled ? disabledStyles : {})}
      {...(checked ? checkedStyles : {})}
    >
      {(indeterminate || checked) && (
        <Icon
          autoid={`${autoid && `${autoid}_`}checkbox_icon`}
          name={indeterminate ? 'minus' : 'check'}
          noFill
          size='full'
        />
      )}
    </Interactable>
  );
};

export default Checkbox;
