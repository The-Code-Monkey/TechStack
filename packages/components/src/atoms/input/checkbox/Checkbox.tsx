import { Interactable, Icon } from '../../../primal';
import { generateAutomationId } from '../../../utils';

import { disabledStyles, checkedStyles } from './styled';

export interface Props {
  testid?: string;
  onClick?: (p1: boolean, p2: MouseEvent) => void;
  checked?: boolean;
  indeterminate?: boolean;
  disabled?: boolean;
  name?: string;
}

const Checkbox = ({
  indeterminate,
  checked,
  onClick,
  disabled,
  testid,
  name,
}: Props) => {
  const handleOnClick = (event: MouseEvent) => {
    const status = indeterminate || !checked;

    if (onClick) {
      onClick(status, event);
    }
  };

  return (
    <Interactable
      name={name}
      display='flex'
      size={7}
      alignItems='center'
      justifyContent='center'
      bgColor='neutrals.0'
      border='1'
      borderRadius='1'
      borderColor='neutrals.10'
      testid={testid ? generateAutomationId(`${testid}_checkbox`) : 'checkbox'}
      data-checked={checked}
      data-indeterminate={indeterminate}
      data-disabled={disabled}
      onClick={handleOnClick}
      {...(disabled ? disabledStyles : {})}
      {...(checked ? checkedStyles : {})}
    >
      {(indeterminate || checked) && (
        <Icon
          testid={
            testid
              ? `${testid}_checkbox_icon`
              : indeterminate
              ? 'minus'
              : 'check'
          }
          name={indeterminate ? 'minus' : 'check'}
          size='full'
        />
      )}
    </Interactable>
  );
};

export default Checkbox;
