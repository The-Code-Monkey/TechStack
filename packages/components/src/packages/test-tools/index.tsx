import { mount, shallow, ShallowWrapper, ReactWrapper } from 'enzyme';
import React, { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';
import { ConfigContext } from '../utils';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: any) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some((prop) => keys.some((key) => key === prop)),
    print: (val: any, serialize: any) => {
      keys.forEach((key) => {
        delete val.props[key];
      });
      return serialize(val);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const shallowWithTheme: (children: ReactNode) => ShallowWrapper = (
  children: ReactNode
) =>
  shallow(<ThemeProvider theme={formattedTheme}>{children}</ThemeProvider>)
    .dive()
    .shallow();

export const mountWithTheme: (children: ReactNode) => ReactWrapper = (
  children: ReactNode
) => {
  expect.addSnapshotSerializer(removeProperties());

  return mount(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>
        <>{children}</>
      </ThemeProvider>
    </ConfigContext.Provider>
  );
};
