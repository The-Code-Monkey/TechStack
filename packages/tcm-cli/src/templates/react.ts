import { PackageJson } from 'type-fest';

import basicTemplate from './basic';
import { Template } from './template';

const reactTemplate: Template = {
  name: 'react',
  dependencies: [
    ...basicTemplate.dependencies,
    '@types/react',
    '@types/react-dom',
    'react',
    'react-dom',
    '@tsconfig/create-react-app',
  ],
  packageJson: {
    ...basicTemplate.packageJson,
    peerDependencies: {
      react: '>=16',
    },
    scripts: {
      ...basicTemplate.packageJson.scripts,
      test: 'tcm test --passWithNoTests',
    } as PackageJson['scripts'],
    jest: {
      testEnvironment: 'jsdom',
    },
  },
};

export default reactTemplate;
