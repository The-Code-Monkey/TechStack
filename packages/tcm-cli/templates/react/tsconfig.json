{
  // see https://www.typescriptlang.org/tsconfig to better understand tsconfigs
  "extends": "../../tsconfig.json",
  "include": ["src", "types"],
"exclude": ["src/**/__test__/*.test.ts*x"],
}
