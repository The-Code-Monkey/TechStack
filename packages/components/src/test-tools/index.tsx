import isPropValid from '@emotion/is-prop-valid';
import { render, RenderResult } from '@testing-library/react';
import { ReactNode } from 'react';
import { StyleSheetManager, ThemeProvider } from 'styled-components';
import { expect } from 'bun:test';

import theme from '../theme';
import { ConfigContext } from '../utils';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: { props: unknown }) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some(prop => keys.some(key => key === prop)),
    serialize: (val, config, indentation, depth, refs, printer) => {
      keys.forEach(key => {
        delete val.props[key];
      });
      return printer(val, config, indentation, depth, refs);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const mountWithTheme = (children: ReactNode): RenderResult => {
  // expect.addSnapshotSerializer(removeProperties());

  return render(
    <StyleSheetManager shouldForwardProp={isPropValid}>
      <ConfigContext.Provider value={config}>
        <ThemeProvider theme={formattedTheme}>
          <>{children}</>
        </ThemeProvider>
      </ConfigContext.Provider>
    </StyleSheetManager>
  );
};
