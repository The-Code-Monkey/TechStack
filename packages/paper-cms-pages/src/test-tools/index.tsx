import { ConfigContext } from '@techstack/components';
import { render } from '@testing-library/react';
import { ReactNode } from 'react';

import theme from '../theme';
import { ThemeProvider } from '../workarounds/styled-components';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: { props: unknown }) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some(prop => keys.some(key => key === prop)),
    serialize: (val, config, indentation, depth, refs, printer) => {
      keys.forEach(key => {
        delete val.props[key];
      });
      return printer(val, config, indentation, depth, refs);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const mountWithTheme = (children: ReactNode) => {
  expect.addSnapshotSerializer(removeProperties());

  return render(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>
        <>{children}</>
      </ThemeProvider>
    </ConfigContext.Provider>
  );
};
