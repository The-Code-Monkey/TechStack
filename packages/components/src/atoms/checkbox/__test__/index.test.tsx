import {waitFor, screen} from "@testing-library/react";
import * as React from 'react';

import { mountWithTheme } from '../../../test-tools';
import Checkbox from '../index';

describe('<Checkbox />', () => {
  it('renders correctly no icon', () => {
    const { asFragment } = mountWithTheme(<Checkbox />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly checked', async () => {
    const { asFragment } = mountWithTheme(<Checkbox checked />);

    await waitFor(() => expect(screen.getByTitle('check-icon')));

    expect(asFragment()).toMatchSnapshot();
  });
  //
  // it('renders correctly indeterminate', () => {
  //   const { asFragment } = mountWithTheme(<Checkbox indeterminate />);
  //   expect(asFragment()).toMatchSnapshot();
  // });
  //
  // it('calls onClick function', async () => {
  //   const {asFragment} = mountWithTheme(<Checkbox />);
  //
  //   expect(asFragment()).toMatchSnapshot()
  //
  //   // await waitFor(() => expect(screen.getByTitle('minus-icon')));
  //
  //   // fireEvent.click(screen.getByTitle('minus-icon'));
  //
  //   expect(asFragment()).toMatchSnapshot()
  //
  //   // await waitFor(() => expect(screen.getByTitle('check-icon')));
  //
  //   expect(asFragment()).toMatchSnapshot()
  // });
});
