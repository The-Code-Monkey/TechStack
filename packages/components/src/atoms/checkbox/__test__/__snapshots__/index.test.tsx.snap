// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders correctly checked 1`] = `
<DocumentFragment>
  <div
    class="sc-bscPHp iWODDL sc-ibERUn"
    cursor="pointer"
    data-autoid="checkbox"
    data-checked="true"
  >
    <div
      class="sc-bscPHp cUnMdB sc-gqyZTE htFnUm"
      data-autoid="undefinedcheckbox_icon_icon"
    >
      <svg
        data-name="check"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          check-icon
        </title>
        <polyline
          points="20 6 9 17 4 12"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Checkbox /> renders correctly no icon 1`] = `
<DocumentFragment>
  <div
    class="sc-bscPHp gTopfE sc-ibERUn"
    cursor="pointer"
    data-autoid="checkbox"
  />
</DocumentFragment>
`;
